4788/src/main/include/Turret2.h

```diff
@@ -3,78 +3,145 @@
 #include "strategy/StrategySystem.h"
 #include "RobotMap.h"
 
+using actState = wml::actuators::BinaryActuatorState;
+
 enum class TurretRotationState {
   IDLE,
   ZEROING,
   MANUAL,
   PID
 };
 
+enum class TurretAngleState {
+  IDLE,
+  ZEROING,
+  MANUAL,
+  PID
+};
+
 enum class TurretFlywheelState {
   IDLE,
   MANUAL,
-  PID,
+  AUTO
 };
 
+
 class Turret : public wml::StrategySystem {
  public:
-  Turret(wml::Gearbox &rotationGearbox,
-         wml::Gearbox &hoodGearbox,
+  Turret(wml::Gearbox &turretRotationGearbox,
+         wml::Gearbox &turretAngleGearbox,
          wml::Gearbox &flywheelGearbox,
-         wml::sensors::BinarySensor &leftLimit,
-         wml::sensors::BinarySensor &angleDownLimit,
-         std::shared_ptr<nt::NetworkTable> &visionTable);
+         wml::sensors::LimitSwitch &rotZeroSensor,
+         wml::sensors::LimitSwitch &angleZeroSensor) :
+         _turretRotationGearbox(turretRotationGearbox),
+         _turretAngleGearbox(turretAngleGearbox),
+         _flywheelGearbox(flywheelGearbox),
+         _rotZeroSensor(rotZeroSensor),
+         _angleZeroSensor(angleZeroSensor) {}
+  
+  // Set the Turrets Subsystems
+  void SetTurretRotation(const TurretRotationState st, double setpoint) {
+    _turretRotationState = st;
+    _rotationSetpoint = setpoint;
+  }
 
-  void SetRotation(const TurretRotationState st, double setpoint) {
-    _rotState = st;
-    _rotSetpoint = setpoint;
+  void SetTurretAngle(const TurretAngleState st, double setpoint) {
+    _turretAngleState = st;
+    _angleSetpoint = setpoint;
   }
 
-  void SetFlywheel(const TurretFlywheelState st, double setpoint) {
-    _flywheelState = st;
+  void SetTurretFlywheel(const TurretFlywheelState st, double setpoint) {
+    _turretFlywheelState = st;
     _flywheelSetpoint = setpoint;
   }
 
-  void UpdateRotation(double dt) {
+  void ZeroTurret() {
+    
+  }
+
+
+  void UpdateTurretRotation(double dt) {
     double voltage = 0;
-    switch(_rotState) {
+    switch (_turretRotationState) {
       case TurretRotationState::IDLE:
         voltage = 0;
-        break;
-      case TurretRotationState::ZEROING:
-        /* ... */
-        break;
+       break;
+
       case TurretRotationState::MANUAL:
-        voltage = 12 * _rotSetpoint;
-        break;
+        voltage = 12 * _rotationSetpoint;
+       break;
+      
       case TurretRotationState::PID:
-        voltage = /* PID LOGIC */;
-        break;
-      /* .. */
+        voltage = 12 * _rotationSetpoint;
+       break;
+      
+      case TurretRotationState::ZEROING:
+        voltage = 12 * _rotationSetpoint;
+       break;
```

Comment:
This should move in a single direction (towards the limit switch) until the limit switch reads true, at which point it should change into `TurretRotationState::MANUAL` with setpoint 0